### PR TITLE
kubeadm: Normalized reset command to match init and join commands.

### PR DESCRIPTION
**Pull request by pires**
**What this PR does / why we need it**: Overall, improves code structure. Opening single PR in order avoid big PRs in the future, when adding tests and new functionality, i.e. #34404.

``` release-note
NONE
```

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35111)

### CODE REVIEW

**Comment by pires** [comment]
@errordeveloper the test failure doesn't seem to be related to this PR. Care to confirm?

**Comment by pires** [comment]
@k8s-bot gci gke e2e test this

File: cmd/kubeadm/app/cmd/reset.go
Comment (errordeveloper) [inline cmd/kubeadm/app/cmd/reset.go]: What is the first return value about exactly?

**Comment by errordeveloper** [comment]
@k8s-bot gci gke e2e test this

File: cmd/kubeadm/app/cmd/reset.go
Comment (pires) [inline cmd/kubeadm/app/cmd/reset.go]: A `Reset` instance.

Comment (errordeveloper) [inline cmd/kubeadm/app/cmd/reset.go]: Ah, it's just a `struct{}`... do we really need it? We don't have dummy structs like this in neither `init.go` nor `join.go`, why is one needed here? Are you planning to add these everywhere, so that we have `init.Run()` and `join.Run()`, etc?

**Comment by pires** [comment]
@k8s-bot gci test this

**Comment by pires** [comment]
It seems GCI e2e is broken:

```
W1019 07:15:51.142] 2016/10/19 07:15:51 e2e.go:709: Step 'teardown' finished in 3m14.747051043s
W1019 07:15:51.143] 2016/10/19 07:15:51 e2e.go:147: Saved XML output to /workspace/_artifacts/junit_runner.xml.
W1019 07:15:51.143] 2016/10/19 07:15:51 e2e.go:179: Something went wrong: starting e2e cluster: error running up: exit status 1
W1019 07:15:51.143] exit status 1
W1019 07:15:51.146] + chmod -R o+r /workspace/_artifacts
W1019 07:15:52.518] + rc=1
W1019 07:15:52.518] + trap '' EXIT
W1019 07:15:52.518] + exit 1
W1019 07:15:52.519] + rc=1
I1019 07:15:52.519] Dumping logs for any remaining nodes
W1019 07:15:52.519] + [[ 1 -ne 0 ]]
W1019 07:15:52.519] + [[ -x cluster/log-dump.sh ]]
W1019 07:15:52.519] + [[ -d _artifacts ]]
W1019 07:15:52.520] + echo 'Dumping logs for any remaining nodes'
W1019 07:15:52.520] + ./cluster/log-dump.sh _artifacts
I1019 07:15:52.544] Dumping master and node logs to _artifacts
I1019 07:15:53.083] Master not detected. Is the cluster up?
I1019 07:15:53.596] Nodes not detected. Is the cluster up?
W1019 07:15:53.596] + [[ 1 -eq 124 ]]
I1019 07:15:53.596] Exiting with code: 1
```

File: cmd/kubeadm/app/cmd/reset.go
Comment (pires) [inline cmd/kubeadm/app/cmd/reset.go]: > Are you planning to add these everywhere, so that we have init.Run() and join.Run(), etc?

Yes, or at least in the cases where there's pre-flight checks. This will make tests easier to implement, as well.

**Comment by pires** [comment]
@k8s-bot gci test this

**Comment by pires** [comment]
@k8s-bot gci test this

**Comment by pires** [comment]
@k8s-bot gci test this

**Comment by pires** [comment]
Any idea what's going on @mikedanese?

**Comment by pires** [comment]
bot gci gke e2e test this

**Comment by pires** [comment]
@k8s-bot gci gke e2e test this

File: cmd/kubeadm/app/cmd/reset.go
Comment (errordeveloper) [inline cmd/kubeadm/app/cmd/reset.go]: > This will make tests easier to implement, as well.
SGTM.

**Review by errordeveloper — APPROVED** [review]
/lgtm

**Comment by errordeveloper** [comment]
LGTM

**Comment by pires** [comment]
@k8s-bot approvers

**Comment by pires** [comment]
@mikedanese @jbeda can any of your unblock this, please?

**Pull request merged**
